### ISSUE TITLE
Sending messages to group chat after number change on the same device

### ISSUE THREAD

**Issue by Mladia** [signalapp/Signal-Android#10865]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I cannot send messages in one particular group chat after I have changed my number. I can still receive messages. All other group chats work.

### Steps to reproduce
- be Admin of a group, make a backup
- register a new number on the same device with the backup from before
- get the new number added to the group
- Try to send to messages to this group

**Actual result:** Strange behavior - on Android I get the symbol under the messages, as if the message is not delivered. 
On a linked Linux desktop device the messages get sent and delivered properly - well I get Partially sent. When I click for more details, I see Signature verification failed, and I see that one of the members has not received the message, but he is also no more active in this group.

**Expected result:** Consistent behavior between Desktop and Android - messages get sent.

### Screenshots

![PC](https://user-images.githubusercontent.com/15912404/106369904-86362f00-6355-11eb-8648-2be725fcf639.png)
![PC2](https://user-images.githubusercontent.com/15912404/106369905-86cec580-6355-11eb-9ff8-d17d7ae23a3d.png)
![signal-2021-01-30-234827](https://user-images.githubusercontent.com/15912404/106369922-bed60880-6355-11eb-999c-d051a0b33015.jpg)

### Device info
**Device:** OnePlus 7t
**Android version:** 10
**Signal version:** 5.3.7

### Link to debug log
https://debuglogs.org/bf1438daa8cba5be74d324a8ae8381bc5e89f564cc980c6aec56dcb7a92e633a

**Comment by colin-axner** [comment]
I ran into this issue as well:
- backup signal
- create new account and restore data
- logout, relogin into old account
- add new account, remove old account, delete old account
- restore data on new account

Others confirm I am in the group, but the group setting says that I am not in the group. I can also see my old number in the group (like it hasn't received the update that I was added). I can receive messages and I can leave the group. Linux client seems to work (though I left before checking so now it just says "unknown group"

I can add a debug log if necessary, though I made the debug log a few hours after running into the issue (forgot to make one at the time)
